### PR TITLE
Update Ocean submit script

### PR DESCRIPTION
**Pull request by Bronoulli**
## Proposed changes

Some changes that will bring Ocean's submit script up to date:
Added Ocean as a machine on develop

Updated Ocean's submit script to be compatible with `spectre schedule`

Removed Clang submit script since Ocean compiles with GCC.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: support/SubmitScripts/Ocean.sh [hunk 1]
```diff
@@ -1,69 +1,17 @@
...
-# Here, -np should take the number of nodes (must be the same as --nodes
-# in the #SBATCH options above).
-SPECTRE_COMMAND="${SPECTRE_EXECUTABLE} +ppn 19 +pemap 0-18 +commap 19"
-
-mpirun -np ${SLURM_JOB_NUM_NODES} --map-by ppr:1:node singularity exec \
-/opt/ohpc/pub/containers/spectre_ocean.sif \
-bash -c "${SPECTRE_COMMAND} --input-file ${SPECTRE_INPUT_FILE}"
+# Ocean is a supercomputer at Cal State, Fullerton.
+# More information:
+# https://github.com/sxs-collaboration/WelcomeToSXS/wiki/Ocean
```
Comment (AlexCarpenter46) [inline support/SubmitScripts/Ocean.sh:8]: This link currently doesn't go anywhere so you can just say it's hosted by Geoffrey Lovelace or add a wiki page for Ocean. That's probably a question for @geoffrey4444 
edit after talking to Geoffrey: "Ocean is a supercomputer at Cal State, Fullerton. For more information, contact Geoffrey Lovelace."

**Review by AlexCarpenter46 — CHANGES_REQUESTED** [review]

**Review by AlexCarpenter46 — APPROVED** [review]

**Comment by AlexCarpenter46** [comment]
Works really well, thanks for changing it to work with the scheduler! :)

File: support/SubmitScripts/Ocean.sh [hunk 2]
```diff
@@ -1,69 +1,17 @@
...
-
-# Choose the executable and input file to run
-# To use an input file in the current directory, set
-# SPECTRE_INPUT_FILE to ${PWD}/InputFileName.yaml
-export SPECTRE_EXECUTABLE=${SPECTRE_BUILD_DIR}/bin/EvolveGeneralizedHarmonic
-export SPECTRE_INPUT_FILE=${PWD}/KerrSchild.yaml
-
-# These commands load the relevant modules and cd into the run directory,
-# creating it if it doesn't exist
-module load ohpc
```
Comment (nilsvu) [inline support/SubmitScripts/Ocean.sh:46]: Is this line not needed anymore?

File: support/SubmitScripts/Ocean.sh [hunk 3]
```diff
@@ -1,69 +1,17 @@
...
-cp ${SPECTRE_INPUT_FILE} ${SPECTRE_RUN_DIR}/
-
-# Set desired permissions for files created with this script
-umask 0022
-
-# Set the path to include the build directory's bin directory
-export PATH=${SPECTRE_BUILD_DIR}/bin:$PATH
-
-# Flag to stop blas in CCE from parallelizing without charm++
-export OPENBLAS_NUM_THREADS=1
```
Comment (nilsvu) [inline support/SubmitScripts/Ocean.sh:60]: And this one

**Review by nilsvu — COMMENTED** [review]
@geoffrey4444 do you want to take a look before we  merge this?

File: support/SubmitScripts/Ocean.sh (hunk 2 above)
Comment (knelli2) [reply support/SubmitScripts/Ocean.sh:46]: We typically don't load modules in the other submit scripts. They have to be loaded on the head node when you submit and they'll carry over.

File: support/SubmitScripts/Ocean.sh [hunk 4]
```diff
@@ -1,69 +1,17 @@
...
-
-# The 19 is there because Charm++ uses one thread per node for communication
-# Here, -np should take the number of nodes (must be the same as --nodes
-# in the #SBATCH options above).
-SPECTRE_COMMAND="${SPECTRE_EXECUTABLE} +ppn 19 +pemap 0-18 +commap 19"
-
-mpirun -np ${SLURM_JOB_NUM_NODES} --map-by ppr:1:node singularity exec \
-/opt/ohpc/pub/containers/spectre_ocean.sif \
-bash -c "${SPECTRE_COMMAND} --input-file ${SPECTRE_INPUT_FILE}"
```
Comment (knelli2) [inline support/SubmitScripts/Ocean.sh:69]: @AlexCarpenter46 or @geoffrey4444 do you know if all this extra core mapping is still necessary? I wouldn't think so, but just to be sure.

**Review by knelli2 — COMMENTED** [review]
Thanks for doing this @Bronoulli!

File: support/SubmitScripts/Ocean.sh (hunk 4 above)
Comment (AlexCarpenter46) [reply support/SubmitScripts/Ocean.sh:69]: I don't think so, I tested it out and it was using the correct number of workers and comm so it seems fine without it

**Review by AlexCarpenter46 — CHANGES_REQUESTED** [review]
I only tested with 1 node, but our summer student tested with multiple nodes and it's not doing what we expect, it's duplicating Executable Output and only utilizing one of the nodes so we'll need to investigate that before merging.

**Review by AlexCarpenter46 — APPROVED** [review]

**Review by knelli2 — APPROVED** [review]

**Review by geoffrey4444 — APPROVED** [review]

**Pull request merged**